test/e2e/framework/log/logger.go

```diff
@@ -59,6 +63,42 @@ func Fail(msg string, callerSkip ...int) {
 	if len(callerSkip) > 0 {
 		skip += callerSkip[0]
 	}
-	log("FAIL", msg)
+	log("FAIL", "%s\n\nFull Stack Trace\n%s", msg, PrunedStack(skip))
 	ginkgowrapper.Fail(nowStamp()+": "+msg, skip)
 }
+
+var codeFilterRE = regexp.MustCompile(`/github.com/onsi/ginkgo/`)
+
+// PrunedStack is a wrapper around debug.Stack() that removes information
+// about the current goroutine and optionally skips some of the initial stack entries.
+// With skip == 0, the returned stack will start with the caller of PruneStack.
+// From the remaining entries it automatically filters out useless ones like
+// entries coming from Ginkgo.
+//
+// This is a copy of https://github.com/onsi/ginkgo/blob/f90f37d87fa6b1dd9625e2b1e83c23ffae3de228/internal/codelocation/code_location.go#L25
+func PrunedStack(skip int) string {
```

Comment:
The original function name is PruneStack in ginkgo. I guess this name (with `d`) is intentional, right?
(I tried to search this name in ginkgo repo, and I missed that with the name)

Reply:
I intentionally renamed it, because this version is simpler and returns the "pruned stack" instead of taking an arbitrary string and pruning it.

Reply:
I added "This is a *modified* copy of PruneStack in ..."